plugins/modules/win_domain_ou.ps1

```diff
@@ -0,0 +1,197 @@
+    }
+
+    # convert to psobject & return
+    return New-Object psobject -Property $parms
+}
+
+Function Get-OuObject {
+    Param([PSObject]$Object)
+    $parms = @{
+        distinguished_name = $Object.DistinguishedName.toString()
+        protected = $Object.ProtectedFromAccidentalDeletion
+    }
+
+    $parms.properties = @{}
+    if ($Object.Created) { $parms.created = $Object.Created.toString() }
+    if ($Object.ObjectGUID) { $parms.guid = $Object.ObjectGUID.toString() }
+    if ($Object.Name) { $parms.name = $Object.Name }
+    if ($Object.Modified) { $parms.modified = $Object.Modified.toString() }
+    if ($Object.ManagedBy) { $parms.properties.managed_by = $Object.ManagedBy }
+    if ($Object.City) { $parms.properties.city = $Object.City }
+    if ($Object.Country) { $parms.properties.country = $Object.Country }
+    if ($Object.PostalCode) { $parms.properties.postal_code = $Object.PostalCode }
+    if ($Object.State) { $parms.properties.state = $Object.State }
+    if ($Object.StreetAddress) { $parms.properties.street_address = $Object.StreetAddress }
+    return $parms | Sort-Object
+}
+
+# attempt import of module
+Try { Import-Module ActiveDirectory }
+Catch { $module.FailJson("The ActiveDirectory module failed to load properly: $($_.Exception.Message)", $_) }
+
+# determine current object state
+Try {
+    $current_ou = Get-ADOrganizationalUnit -Filter * -Properties * | Where-Object {
+        $_.DistinguishedName -eq "OU=$name,$path"
+    }
+    $module.Diff.before = Get-OuObject -Object $current_ou
+    $module.Result.ou = $module.Diff.before
+} Catch {
+    $module.Diff.before = ""
+    $current_ou = $false
+}
+
+if ($state -eq "present") {
+    # parse inputs
+    if ($properties.city) { $parms.City = $properties.city }
+    if ($properties.description) { $parms.Description = $properties.description }
+    if ($properties.display_name) { $parms.DisplayName = $properties.display_name }
+    if ($properties.country) { $parms.Country = $properties.country }
+    if ($properties.managed_by) { $parms.ManagedBy = $properties.managed_by }
+    if ($properties.street_address) { $parms.StreetAddress = $properties.street_address }
+    if ($properties.state) { $parms.State = $properties.state }
+
+    # ou does not exist, create object
+    if(-not $current_ou) {
+        $parms.Name = $name
+        $parms.Path = $path
+        Try { New-ADOrganizationalUnit @parms -ProtectedFromAccidentalDeletion $protected -WhatIf:$check_mode }
+        Catch { $module.FailJson("Failed to create organizational unit: $($_.Exception.Message)") }
+    }
+
+    # ou exists, update object
+    if ($current_ou) {
+        Try { Set-ADOrganizationalUnit -Identity "OU=$name,$path" @parms -WhatIf:$check_mode }
+        Catch {
+            $module.Result.debug = $parms
+            $module.FailJson("Failed to update organizational unit: $($_.Exception.Message)") }
```

Comment:
Should add `$_` as a 2nd argument to `FailJson` so the stacktrace isn't lost when using a higher verbosity in Ansible.